src/dos/drive_local.cpp

```diff
@@ -19,7 +19,10 @@
  *  With major works from joncampbell123 and Wengier
  */
 
+#if defined(HAIKU)
 #include "config.h"
```

Comment:
This was added exactly because otherwise ``HAIKU`` wasn't defined, so probably the right fix would be to remove the ``config.h`` and understand why ``HAIKU`` isn't defined but the other OSes are

Reply:
https://github.com/joncampbell123/dosbox-x/pull/4002 should fix MinGW without breaking HAIKU

Reply:
@amol- Thanks for your confirmation.
 I see that MinGW build is back with your fix  so I will withdraw this request.